internal/controller/platform/standalone/nimservice.go

```diff
@@ -139,89 +146,175 @@ func (r *NIMServiceReconciler) reconcileNIMService(ctx context.Context, nimServi
+	modelProfile := nimService.GetNIMCacheProfile()
 	if modelProfile != "" {
+		// Set environment variable for model profile
 		profileEnv := corev1.EnvVar{
 			Name:  "NIM_MODEL_PROFILE",
 			Value: modelProfile,
 		}
 		deploymentParams.Env = append(deploymentParams.Env, profileEnv)
 
-		// TODO: assign GPU resources and node selector that is required for the selected profile
-		// TODO: assign GPU resources
-		// TODO: update the node selector
-	}
+		// Retrieve and set profile details from NIMCache
+		profile, err := r.getNIMCacheProfile(ctx, nimService, modelProfile)
+		if err != nil {
+			logger.Error(err, "Specified profile not found in NIMCache")
+			return err
+		}
 
-	// Sync deployment
-	err = r.renderAndSyncResource(ctx, nimService, &renderer, &appsv1.Deployment{}, func() (client.Object, error) {
-		return renderer.Deployment(deploymentParams)
-	}, "deployment", conditions.ReasonDeploymentFailed)
-	if err != nil {
-		return ctrl.Result{}, err
+		// Assign GPU resources and update node selector based on the profile
+		if err := r.assignResourcesAndNodeSelector(ctx, nimService, profile, deploymentParams); err != nil {
+			return err
+		}
 	}
 
-	// Wait for deployment
-	msg, ready, err := r.isDeploymentReady(ctx, &namespacedName)
+	return nil
+}
+
+func (r *NIMServiceReconciler) assignResourcesAndNodeSelector(ctx context.Context, nimService *appsv1alpha1.NIMService, profile *appsv1alpha1.NIMProfile, deploymentParams *rendertypes.DeploymentParams) error {
+	logger := log.FromContext(ctx)
+
+	// Assign GPU resources
+	tensorParallelism, err := r.getTensorParallelismByProfile(ctx, profile)
 	if err != nil {
-		return ctrl.Result{}, err
+		logger.Error(err, "Failed to retrieve tensorParallelism")
+		return err
 	}
 
-	if !ready {
-		// Update status as NotReady
-		err = r.updater.SetConditionsNotReady(ctx, nimService, conditions.NotReady, msg)
-	} else {
-		// Update status as ready
-		err = r.updater.SetConditionsReady(ctx, nimService, conditions.Ready, msg)
+	// Check if tensorParallelism is provided and assign GPU resources
+	if tensorParallelism != "" {
+		gpuQuantity, err := apiResource.ParseQuantity(tensorParallelism)
+		if err != nil {
+			return fmt.Errorf("failed to parse tensorParallelism: %w", err)
+		}
+
+		// Ensure that the Resources field is initialized
+		if deploymentParams.Resources == nil {
+			deploymentParams.Resources = &corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{},
+				Limits:   corev1.ResourceList{},
+			}
+		}
+
+		// TODO: Make the resource name configurable
+		const gpuResourceName = corev1.ResourceName("nvidia.com/gpu")
+
+		deploymentParams.Resources.Requests[corev1.ResourceName(gpuResourceName)] = gpuQuantity
+		deploymentParams.Resources.Limits[corev1.ResourceName(gpuResourceName)] = gpuQuantity
 	}
 
+	// Apply node selector based on the device ID from the profile
+	// TODO: handle OCP specific logic for NFD API group
+	deviceID, err := r.getDeviceIDByProfile(ctx, profile)
```

Comment:
Wondering if a model profile can be run with multiple possible gpu types, instead of one type